### DISCUSSION TITLE
Cannot run yarn dev

### DISCUSSION THREAD

**Discussion by osseonews** [graphcommerce-org/graphcommerce#1377] [Q&A] [answered]
We installed the example package locally and ran: yarn, and then yarn dev. When we visit localhost:3000 we get the error below. Any ideas? 

```
Module not found: Can't resolve './generated/types'
> 1 | export * from './generated/types'
  2 | 
  3 | export * from './measurePerformanceLink'
  4 | export * from './errorLink'

Import trace for requested module:
./lib/graphql/GraphQLProvider.tsx
./pages/_app.tsx
```

**Comment by paales** [comment, marked as answer]
@osseonews You are almost there! Before you run `yarn dev` you first need to run `yarn codegen` as mentioned here: https://www.graphcommerce.org/docs/getting-started/create#step-3-start-the-development-environment

**Reply by swapnilbijwe-rc** [reply]
t] https://nextjs.org/docs/messages/module-not-found
[next] wait  - compiling...
[next] error - ./node_modules/@graphcommerce/graphql/components/GraphQLProvider.tsx:14:0
[next] Module not found: Can't resolve '../generated/fragments.json'
[next] 
[next] Import trace for requested module:
[next] ./node_modules/@graphcommerce/graphql/index.ts
[next] ./node_modules/@graphcommerce/graphql/index.interceptor.tsx
[next] ./pages/_app.tsx
[next] 
[next] https://nextjs.org/docs/messages/module-not-found

**Reply by paales** [reply]
What is the output when you run `yarn codegen`?

**Comment by swapnilbijwe-rc** [comment]
***@***.******@***.***> graphcommerce-poc % yarn graphql-codegen

yarn run v1.22.19

$ /Users/swapnil.b/Sites/graphcommerce/graphcommerce-poc/node_modules/.bin/graphql-codegen

✔ Parse Configuration

✔ Parse Configuration

⚠ Generate outputs

  ✔ Generate to ***@***.******@***.***/graphql/generated/types.ts>

  ❯ Generate to ***@***.******@***.***/graphql/generated/fragments.json>

    ✔ Load GraphQL schemas

    ✔ Load GraphQL documents

    ✖ Interface field CartItemInterface.errors expected but GiftCardCartItem does not provide it.

  ❯ Generate to .

    ✔ Load GraphQL schemas

    ✔ Load GraphQL documents

    ✖ Fragment "BlogAuthor" is set on non-existing type "Page"!

error Command failed with exit code 1.

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

***@***.******@***.***> graphcommerce-poc % yarn dev


Swapnil Bijwe

[signature_4025508282]
Practice Lead(Adobe)
Cell: +91 8600900696
***@***.******@***.***>
www.royalcyber.com<https://nam10.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.royalcyber.com%2F&data=02%7C01%7Cwasey.ameen%40royalcyber.com%7C59e51a6277d34ce2881108d85673fb0c%7Cd508624fa0b74fd3951105b18ca02784%7C1%7C0%7C637354403431128646&sdata=2uBtVkJlx63IakkrAIqGwdd2IoxiodowPlEeJP0XK%2Fw%3D&reserved=0>



From: Paul Hachmang ***@***.***>
Date: Wednesday, 25 January 2023 at 8:03 PM
To: graphcommerce-org/graphcommerce ***@***.***>
Cc: Swapnil Bijwe ***@***.***>, Comment ***@***.***>
Subject: [EXTERNAL]Re: [graphcommerce-org/graphcommerce] Cannot run yarn dev (Discussion #1377)

What is the output when you run yarn codegen?

—
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgraphcommerce-org%2Fgraphcommerce%2Fdiscussions%2F1377%23discussioncomment-4778244&data=05%7C01%7Cswapnil.bijwe%40royalcyber.com%7C7d03901c8ffe410b576508dafee11139%7Cd508624fa0b74fd3951105b18ca02784%7C1%7C0%7C638102539875964399%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=8nEZm6KUdLSpCpsnb28eRMjSiVlqw7XvQu%2BfAhywZqc%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FA5C74GOXRLPAAZMKNPMEWSLWUE2R7ANCNFSM5SNT3CKQ&data=05%7C01%7Cswapnil.bijwe%40royalcyber.com%7C7d03901c8ffe410b576508dafee11139%7Cd508624fa0b74fd3951105b18ca02784%7C1%7C0%7C638102539875964399%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=HhgoCh%2FXJetqnxXn8pxKYSdybrxUFW6OwrUwVvCOfkI%3D&reserved=0>.
You are receiving this because you commented.Message ID: ***@***.***>

Disclaimer: This message and any files transmitted with it are confidential and privileged. If you have received it in error, please notify the sender by return e-mail and delete this message from your system. If you are not the intended recipient you are hereby notified that any dissemination, copy or disclosure of this e-mail is strictly prohibited.

**Reply by paales** [reply]
Ohh I seee.. You are running Adobe Commerce 2.4.3 it seems, but there seems to be an incompatibility between Adobe Commerce 2.4.3 and Adobe Commerce. In an ideal scenario I'd ask you to upgrade to Magento 2.4.5 but that might not be an option right now.

Workaround for now:

Can you create a file `graphql/FixCommerce243.graphqls`
```graphqls
type GiftCardCartItem {
  "An array of errors encountered while loading the cart item"
  errors: [CartItemError]
}
```

And after that run `yarn codegen` again.
